### PR TITLE
Bump docker image to use stable poetry version

### PR DESCRIPTION
**Pull request by erikjohnston**
Seems like a sensible thing, but also makes my life slightly easier in the rust PR.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -46,17 +46,8 @@ RUN \
...
-#    https://github.com/python-poetry/poetry/pull/5156 and
-#    https://github.com/python-poetry/poetry/issues/5141 ;
-# without it, we generate a requirements.txt with incorrect environment markers,
-# which causes necessary packages to be omitted when we `pip install`.
-#
-# NB: In poetry 1.2 `poetry export` will be moved into a plugin; we'll need to also
-# pip install poetry-plugin-export (https://github.com/python-poetry/poetry-plugin-export).
 RUN --mount=type=cache,target=/root/.cache/pip \
-  pip install --user "poetry-core==1.1.0a7" "git+https://github.com/python-poetry/poetry.git@fb13b3a676f476177f7937ffa480ee5cff9a90a5"
+  pip install --user poetry>=1.2.0
```
Comment (DMRobertson) [inline docker/Dockerfile:50]: Why a lower bound rather than a pin?

Comment (erikjohnston) [reply docker/Dockerfile:50]: Oh, yeah, a pin is probably better

**Review by DMRobertson — COMMENTED** [review]
For paranoia's sake, I'd like to see a diff of `pip list` in the docker container's python environment before and after this change.

**Comment by erikjohnston** [comment]
Given that we run sytest and complement against the docker image I'm fairly confident CI should pick things up. But may as well double check the `pip list` anyway:

<details>
<summary>Pip list</summary>

```
Package              Version
-------------------- ---------
attrs                21.4.0
Authlib              0.15.5
Automat              20.2.0
bcrypt               3.2.0
bleach               4.1.0
canonicaljson        1.6.0
certifi              2021.10.8
cffi                 1.15.0
charset-normalizer   2.0.12
constantly           15.1.0
cryptography         36.0.1
defusedxml           0.7.1
elementpath          2.5.0
frozendict           2.3.3
hiredis              2.0.0
hyperlink            21.0.0
idna                 3.3
ijson                3.1.4
incremental          21.3.0
jaeger-client        4.8.0
Jinja2               3.0.3
jsonschema           4.4.0
ldap3                2.9.1
lxml                 4.9.1
MarkupSafe           2.1.0
matrix-common        1.2.1
matrix-synapse       1.66.0
matrix-synapse-ldap3 0.2.2
msgpack              1.0.3
netaddr              0.8.0
opentracing          2.4.0
packaging            21.3
parameterized        0.8.1
phonenumbers         8.12.44
Pillow               9.0.1
pip                  22.0.4
prometheus-client    0.14.0
psycopg2             2.9.3
pyasn1               0.4.8
pyasn1-modules       0.2.8
pycparser            2.21
pydantic             1.9.1
pymacaroons          0.13.0
Pympler              1.0.1
PyNaCl               1.5.0
pyOpenSSL            22.0.0
pyparsing            3.0.7
pyrsistent           0.18.1
pysaml2              7.1.2
python-dateutil      2.8.2
pytz                 2021.3
PyYAML               6.0
requests             2.27.1
sentry-sdk           1.5.11
service-identity     21.1.0
setuptools           58.1.0
signedjson           1.1.4
simplejson           3.17.6
six                  1.16.0
sortedcontainers     2.4.0
threadloop           1.0.2
thrift               0.15.0
tornado              6.1
treq                 22.2.0
Twisted              22.4.0
txredisapi           1.4.7
typing_extensions    4.1.1
unpaddedbase64       2.1.0
urllib3              1.26.8
webencodings         0.5.1
wheel                0.37.1
xmlschema            1.10.0
zope.interface       5.4.0
```

</details>

**Comment by DMRobertson** [comment]
> Given that we run sytest and complement against the docker image I'm fairly confident CI should pick things up.

We'd pick up bugs, sure, but I'm wanting to check that `poetry export` still behaves like we want it to. In any case, the `pip list` output LGTM!

File: docker/Dockerfile [hunk 2]
```diff
@@ -46,17 +46,8 @@ RUN \
...
 # synapse's dependencies.
-# We use a specific commit from poetry's master branch instead of our usual 1.1.14,
-# to incorporate fixes to some bugs in `poetry export`. This commit corresponds to
-#    https://github.com/python-poetry/poetry/pull/5156 and
-#    https://github.com/python-poetry/poetry/issues/5141 ;
-# without it, we generate a requirements.txt with incorrect environment markers,
-# which causes necessary packages to be omitted when we `pip install`.
-#
-# NB: In poetry 1.2 `poetry export` will be moved into a plugin; we'll need to also
-# pip install poetry-plugin-export (https://github.com/python-poetry/poetry-plugin-export).
```
Comment (clokep) [inline docker/Dockerfile:57]: Is this not true -- do we not need this plug-in?

Comment (erikjohnston) [reply docker/Dockerfile:57]: Nope, poetry 1.2 pulls in the export plugin by default

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -283,7 +283,6 @@ jobs:
       - run: sudo apt-get -qq install xmlsec1
       - uses: matrix-org/setup-python-poetry@v1
         with:
-          python-version: ${{ matrix.python-version }}
```
Comment (DMRobertson) [inline .github/workflows/tests.yml:286]: Just a typo I guess?

**Review by DMRobertson — APPROVED** [review]
Works for me. (I'm glad you're doing this rather than me...)

**Pull request merged**